cg/apps/demultiplex/sample_sheet/models.py

```diff
@@ -76,6 +78,15 @@ class FlowCellSampleNovaSeq6000Dragen(FlowCellSampleNovaSeq6000):
 class SampleSheet(BaseModel):
     samples: List[FlowCellSample]
 
+    def get_non_pooled_samples(self) -> List[FlowCellSample]:
+        """Return samples that are sequenced solo in their lane."""
+        lane_samples = defaultdict(list)
+        for sample in self.samples:
+            lane: int = sample.lane
+            lane_samples[lane].append(sample)
```

Comment:
I think duplication of this kind is fine since it is just three lines of code. It might end up more confusing calling the helper function. What do you think?

Reply:
I think the usages can be fixed so that the function is defined only once. But that is out of scope. I will create an issue and create a tag `demultiplex` to group all the OOS ideas regarding refactoring the demultiplex module